File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
'''
U - identify the frequency of integers in an array. return the top k frequent elements
    - I: int[] - an array of integers
    - O: int[] - an array of the top two elemts
    - E: 
        if arr.length == 1 return arr
        
M - Frequency Mapping
P - using a dictionary

    # create dictionary

    # check if number exists in dict

        # if yes, increase count

        # if no, instantiate the value
    
    # return a list of the highest frequencies (dict.items())
    # sort the list DESC

    # create array which stores frequencies [:k]

    # create another array (top_nums[])
    for i in new_array:
        top_nums.add(dict.get((i)))

    return top_nums
I -
R -
E -
'''


class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:    
        # create dictionary
        freq = {}
        # check if number exists in dict
        for num in nums:
            if num in freq:
            # if yes, increase count
                freq[num] = freq[num] + 1
            # if no, instantiate the value
            else:
                freq[num] = 1
        print(freq)
        # create a list of the highest frequencies (dict.items())
        # sort the list DESC
        freq_values = list(freq.values())
        print(freq_values.sort(reverse=True))
        # create array which stores frequencies [:k]
        freq_values = freq_values[:k]
        print(freq_values)

        # create another array (top_nums[])
        top_nums = []
        for key, val in freq.items():
            if val in freq_values:
                top_nums.append(key)

        return top_nums[:k]
        

File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
       let dub = false
       let count = 1
       const map = new Map()

       for (let i of nums) {
            if (map.has(i)){
               count = map.get(i) + 1;
               map.set(i, count);
               console.log(map)
            }
            else {
                map.set(i, count)
            }
       }

       for (let [key, value] of map){
        console.log(value)
        if(value > 1)
            dub = true
       }

       return dub;

    }
}
